src/frontend/src/Components/map/panels/CamPanel.js

```diff
@@ -76,6 +78,9 @@ export default function CamPanel(props) {
       return;
     }
     setCamera(rootCam.camGroup[camIndex]);
+    const url = new URL(window.location);
+    url.searchParams.set("camIndex", camIndex);
+    window.history.replaceState({}, "", url);
```

Comment:
Please use const [searchParams, setSearchParams] = useSearchParams(); in this file instead of manually setting and getting.